ui/ducks/send/send.js

```diff
@@ -0,0 +1,435 @@
+import { createSlice } from '@reduxjs/toolkit';
+import abi from 'human-standard-token-abi';
+import log from 'loglevel';
+import { addHexPrefix } from 'ethereumjs-util';
+import { multiplyCurrencies } from '../../helpers/utils/conversion-util';
+import { GAS_LIMITS } from '../../../shared/constants/gas';
+import { MIN_GAS_LIMIT_HEX } from '../../pages/send/send.constants';
+
+import {
+  addGasBuffer,
+  calcGasTotal,
+  calcTokenBalance,
+  generateTokenTransferData,
+} from '../../pages/send/send.utils';
+import {
+  getAveragePriceEstimateInHexWEI,
+  getSelectedAccount,
+  getTargetAccount,
+} from '../../selectors';
+import { estimateGas } from '../../store/actions';
+import { setCustomGasLimit } from '../gas/gas.duck';
+
+const name = 'send';
+
+export const initialState = {
+  gasButtonGroupShown: true,
+  gasLimit: null,
+  gasPrice: null,
+  gasTotal: null,
+  gasIsLoading: false,
+  tokenBalance: '0x0',
+  from: '',
+  to: '',
+  amount: '0',
+  memo: '',
+  errors: {},
+  maxModeOn: false,
+  editingTransactionId: null,
+  toNickname: '',
+  ensResolution: null,
+  ensResolutionError: '',
+};
+
+const slice = createSlice({
+  name,
+  initialState,
+  reducers: {
+    updateSendErrors: (state, action) => {
+      state.errors = { ...state.errors, ...action.payload };
+    },
+    showGasButtonGroup: (state) => {
+      state.gasButtonGroupShown = true;
+    },
+    hideGasButtonGroup: (state) => {
+      state.gasButtonGroupShown = false;
+    },
+    setGasLimit: (state, action) => {
+      state.gasLimit = action.payload;
+    },
+    setGasPrice: (state, action) => {
+      state.gasPrice = action.payload;
+    },
+    setSendTokenBalance: (state, action) => {
+      state.tokenBalance = action.payload;
+    },
+    updateSendHexData: (state, action) => {
+      state.data = action.payload;
+    },
+    updateSendTo: (state, action) => {
+      state.to = action.payload.to;
+      state.toNickname = action.payload.nickname;
+    },
+    updateSendAmount: (state, action) => {
+      state.amount = action.payload;
+    },
+    setMaxModeTo: (state, action) => {
```

Comment:
Nit: The naming conventions now seem a bit inconsistent - the "setter" and "update" methods are both kinda the same. Maybe we could normalize to use "update" or "set" consistently?

Maybe better to use "update" so people don't confuse this for a "setter" in the bad sense. Since using getters/setters in a Redux slice is generally a bad pattern - instead we should use events that represents the actual thing that happened in the world, e.g. the user action. The exception being cases like this, where the actual event that occurred was a field being changed.

Reply:
this should now be reflected in the work, although it has changed drastically since this comment :P